### PR TITLE
Add config for vault secret collection manager

### PR DESCRIPTION
**Pull request by alvaroaleman**
Ref https://issues.redhat.com/browse/DPTP-2068
/cc @openshift/openshift-team-developer-productivity-test-platform

### CODE REVIEW

File: clusters/app.ci/vault/secret-collection-manager.yaml [hunk 1]
```diff
@@ -1,34 +1,18 @@
-apiVersion: image.openshift.io/v1
-kind: ImageStream
+apiVersion: v1
```
Comment (alvaroaleman) [inline clusters/app.ci/vault/secret-collection-manager.yaml:1]: I accidentally included a WIP version of this in b5defe31109213e375bb8cc6c0626f78515fb7de, hence the diff

File: hack/deploy_vault_config.sh [hunk 2]
```diff
@@ -200,6 +200,28 @@ getUserIDByLDAPName() {
 getUserIDByLDAPName brawilli
 vault write identity/group name="release-controller" policies="release-controller" member_entity_ids="$(getUserIDByLDAPName brawilli)"
 
+vault policy write vault-secret-collection-manager -<<EOH
```
Comment (alvaroaleman) [inline hack/deploy_vault_config.sh:203]: Something with these is still wrong, but its hard to test when the deployment continously overwrites :upside_down_face:

File: core-services/ci-secret-bootstrap/_config.yaml [hunk 3]
```diff
@@ -2979,3 +2979,25 @@ secret_configs:
...
+      bw_item: dex
+      field: vault-secret-manager-client-id
+    vault-secret-manager-secret:
+      bw_item: dex
+      field: vault-secret-manager-client-secret
+  to:
+  - cluster: app.ci
+    name: vault-secret-manager
+    namespace: dex
+  - cluster: app.co
```
Comment (hongkailiu) [inline core-services/ci-secret-bootstrap/_config.yaml:2993]:
```suggestion
  - cluster: app.ci
```

**Review by hongkailiu — APPROVED** [review]

**Pull request merged**